src/components/Common/ReorderableList/ReorderableItem.tsx

```diff
@@ -0,0 +1,239 @@
+import { useRef, useEffect, useCallback, memo } from 'react'
+import classnames from 'classnames'
+import { useTranslation } from 'react-i18next'
+import { useDrag } from 'react-dnd'
+import type { ReactElement } from 'react'
+import { VisuallyHidden } from '../VisuallyHidden'
+import type { ReorderableListItem } from './ReorderableListTypes'
+import styles from './ReorderableList.module.scss'
+import { ITEM_TYPE } from './constants'
+import ListIcon from '@/assets/icons/list.svg?react'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+
+interface ReorderableItemProps {
+  item: ReorderableListItem
+  index: number
+  moveItem: (fromIndex: number, toIndex: number, source?: 'keyboard' | 'dragdrop') => void
+  itemCount: number
+  itemIndex: number
+  listId: string
+  isDraggingAny: boolean
+  setIsDragging: (isDragging: boolean) => void
+  isInitialized: boolean
+  isReorderingActive: boolean
+  setIsReorderingActive: (isReorderingActive: boolean) => void
+  isCurrentlyReordering: boolean
+  setReorderingItemIndex: (reorderingItemIndex: number | null) => void
+  renderDragHandle?: (props: {
+    id: string | number
+    label: string
+    isReordering: boolean
+    isDragging: boolean
+  }) => ReactElement
+  className?: string
+}
+
+/**
+ * Component for an individual reorderable item
+ */
+export const ReorderableItem = memo(function ReorderableItem({
+  item,
+  index,
+  moveItem,
+  itemCount,
+  itemIndex,
+  listId,
+  isDraggingAny,
+  setIsDragging,
+  isInitialized,
+  isReorderingActive,
+  setIsReorderingActive,
+  isCurrentlyReordering,
+  setReorderingItemIndex,
+  renderDragHandle,
+  className,
+}: ReorderableItemProps) {
+  const ref = useRef<HTMLDivElement>(null)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+  const { t } = useTranslation('common')
+  const Components = useComponentContext()
+
+  // Format the accessible item name
+  const accessibleItemName = item.label
+
+  // Add a ref to track if this specific item is being dragged
+  const isBeingDraggedRef = useRef(false)
+
+  // When this item becomes the reordering item, focus on it
+  useEffect(() => {
+    if (isCurrentlyReordering && buttonRef.current) {
+      buttonRef.current.focus()
+    }
+  }, [isCurrentlyReordering])
+
+  // React DnD - Drag source
+  const [{ isDragging }, drag, dragPreview] = useDrag(
+    () => ({
+      type: ITEM_TYPE,
+      item: () => {
+        isBeingDraggedRef.current = true
+        setIsDragging(true)
+        return { index, listId }
+      },
+      collect: monitor => ({
+        isDragging: !!monitor.isDragging(),
+      }),
+      end: () => {
+        isBeingDraggedRef.current = false
+
+        setTimeout(() => {
+          setIsDragging(false)
+        }, 50)
```

Comment:
Same here, should we consider clearing this timeout?

Reply:
Fixed